contracts/Remittance.sol

```diff
@@ -1,16 +1,138 @@
 //SPDX-License-Identifier: MIT
 pragma solidity >= 0.6.0 < 0.7.0;
 
-import "../node_modules/@openzeppelin/contracts/access/Ownable.sol";
-import "../node_modules/@openzeppelin/contracts/utils/Pausable.sol";
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
 
-import {SafeMath} from "../node_modules/@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeMath} from "./math/SafeMath.sol";
 
 contract Remittance is Ownable, Pausable {
 
-    using SafeMath for uint;
+    using SafeMath for uint256;
 
+    mapping(bytes32 => RemittanceInstance) public remittances;
 
+    // The 'valid' element provides a flag for checking a bonafide hash key
+    struct RemittanceInstance {
+        address funder;
+        address broker;
+        uint256 fundsOwed;
+        uint256 expiryDate;
+        bool valid;
+    }
+
+    event RemittanceFundsCreated(
+        bytes32 indexed hash,
+        address indexed funder,
+        address indexed broker,
+        uint256 amount,
+        uint256 expiryDate
+    );
+
+    event RemittanceFundsReleased(
+        bytes32 indexed hash,
+        address indexed broker,
+        uint256 amount
+    );
+
+    event RemittanceFundsReclaimed(
+        bytes32 indexed hash,
+        address indexed funder,
+        uint256 amount
+    );
+
+    // The funder can create a remittance
+    // Hash is a sha256 hash of the broker address + recipient password
+    function create(bytes32 recipientPasswordHash, address broker, uint256 expiryDate)
+    external
+    payable
+    whenNotPaused
+    returns(bool success)
+    {
+        require(broker != msg.sender, "The caller cannot be the broker");
+        require(msg.value > 0, "The amount must be greater than 0");
+        require(broker != address(0), "Address cannot be zero");
+        checkEmptyHash(recipientPasswordHash);
+
+        RemittanceInstance storage remittanceInstance = remittances[recipientPasswordHash];
+
+        require(remittanceInstance.funder == address(0), "Remittance already exists");
+
+        remittanceInstance.funder = msg.sender;
+        remittanceInstance.broker = broker;
+        remittanceInstance.fundsOwed = msg.value;
+        remittanceInstance.expiryDate = expiryDate;
+        remittanceInstance.valid = true;
```

Comment:
Is this `.valid` field absolutely necessary?

Reply:
I put reasoning for .valid here:

https://github.com/onshop/remittance/pull/1#discussion_r589086644

Perhaps over-engineered. I'll remove it.

Reply:
I have removed.